content/oci/storage.go

```diff
@@ -106,6 +106,22 @@ func (s *Storage) Push(_ context.Context, expected ocispec.Descriptor, content i
 	return nil
 }
 
+// Delete removes the descriptor blob from the storage
+func (s *Storage) Delete(ctx context.Context, target ocispec.Descriptor) error {
+	path, err := blobPath(target.Digest)
+	if err != nil {
+		return fmt.Errorf("%s: %s: %w", target.Digest, target.MediaType, errdef.ErrInvalidDigest)
+	}
+	fullpath := filepath.Join(s.root, path)
+
+	// check if the target content does not exists in the blob directory.
+	if _, err := os.Stat(fullpath); os.IsNotExist(err) {
+		return err
+	}
```

Comment:
There is a pitfall in `os.Stat()`. A bug free implementation would be 
```suggestion
	// check if the target content does not exists in the blob directory.
	if _, err := os.Stat(fullpath); err != nil {
		if os.IsNotExist(err) {
			// return not exist error
		}
		// return generic error
	}
```

Reply:
Since `os.Remove()` checks the file existence, we don't need to do `os.Stat()` in advance.